p2p/multiplexer.py

```
@@ -295,15 +295,15 @@ def stream_protocol_messages(self,
             token = self._multiplex_token
 
             while not self.is_closing and not token.triggered:
-                try:
-                    # We use an optimistic strategy here of using
-                    # `get_nowait()` to reduce the number of times we yield to
-                    # the event loop.  Since this is an async generator it will
-                    # yield to the loop each time it returns a value so we
-                    # don't have to worry about this blocking other processes.
+                # We use an optimistic strategy here of using
+                # `get_nowait()` to reduce the number of times we yield to
+                # the event loop.  Since this is an async generator it will
+                # yield to the loop each time it returns a value so we
+                # don't have to worry about this blocking other processes.
+                if msg_queue.empty():
+                    yield await msg_queue.get()
+                else:
```

This is the change I made to fix the weird CI error.  the functionality should be the same but it no longer ignores an exception async iteration which appeared to be part of the problem, but it wasn't clear exactly why....